### PR TITLE
[STG93] Adding Rename Support to List Ranges API

### PR DESCRIPTION
**Pull request by tasherif-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTestFixtureAttribute.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public ShareClientTestFixtureAttribute(params object[] additionalParameters)
                 additionalParameters: additionalParameters)
         {
             RecordingServiceVersion = StorageVersionExtensions.MaxVersion;
-            LiveServiceVersions = new object[] { StorageVersionExtensions.LatestVersion, };
+            LiveServiceVersions = new object[] { StorageVersionExtensions.MaxVersion, };
```
Comment (tasherif-msft) [inline sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTestFixtureAttribute.cs:44]: TODO: revert

File: sdk/storage/Azure.Storage.Files.Shares/src/Generated/Models/ShareErrorCode.cs [hunk 2]
```diff
@@ -207,6 +208,8 @@ public ShareErrorCode(string value)
         public static ShareErrorCode ShareSnapshotOperationNotSupported { get; } = new ShareErrorCode(ShareSnapshotOperationNotSupportedValue);
         /// <summary> ShareHasSnapshots. </summary>
         public static ShareErrorCode ShareHasSnapshots { get; } = new ShareErrorCode(ShareHasSnapshotsValue);
+        /// <summary> PreviousSnapshotNotFound. </summary>
+        public static ShareErrorCode PreviousSnapshotNotFound { get; } = new ShareErrorCode(PreviousSnapshotNotFoundValue);
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/Generated/Models/ShareErrorCode.cs:212]: I think you might need to run `eng/scripts/ExportApi.ps1 Storage` again, this new error code should appear in sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.net6.0.cs

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/Generated/Models/ShareErrorCode.cs:212]: Also, where did you discover this new error code?  In the spec, or was it returned by the service?

Comment (tasherif-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/Generated/Models/ShareErrorCode.cs:212]: it was returned by the service in the error case (409)!
I will run ps1 again

**Comment by seanmcc-msft** [comment]
Looks like there was an issue with recording the new test.

**Comment by tasherif-msft** [comment]
> Looks like there was an issue with recording the new test.

yea i ran live earlier, should be all fixed now :)

**Review by seanmcc-msft — APPROVED** [review]
Approved, but hold off merging for now.

**Pull request merged**